capstone/src/main/java/com/google/sps/servlets/LessonHandlerServlet.java

```diff
@@ -64,15 +65,44 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     public List<Lesson> getLessons(Room room) {
         List<Lesson> lessons = new ArrayList<>();
+        List<Key> nullKeys = new ArrayList<>();
     
         try {
             for(Key key : room.getAllLessons()) {
-                lessons.add(DatabaseService.getLesson(key));
+                Boolean status = checkLessonStatus(key);
+
+                if (status) {
+                    lessons.add(DatabaseService.getLesson(key));
+                } else {
+                    nullKeys.add(key);
+                }
             }
         }
         catch (EntityNotFoundException e) {
             System.err.println("Lesson entities don't exist.");
         }
+        removeDeletedLessons(room, nullKeys);
         return lessons;
     }
```

Comment:
This method should probably live in the DatabaseService, since it could be re-used by any other part of the code which needs a full list of Lessons from a room.